### ISSUE TITLE
This extension is not installable on any currently installed products

### ISSUE THREAD

**Issue by BOBJohnson** [gitextensions/gitextensions.vsix#20]
I get the following error when trying to install the VSIX:

"This extension is not installable on any currently installed products."

I had to rollback my VS 2019 recently because VS became unusable (it would take 15+ minutes to open a single project solution, and anything more complex just hangs indefinitely). I don't know what version I was on before VS decided to become unusable - plus I'm not even sure it's was VS's fault (definitely was MS's fault) as when I noticed the issue was after a Windows update for the .Net framework.

Anyway, I'm now on version 16.8.0 because I just picked an older version at random cause I wasn't sure rolling back would even work, nor what version would. However 16.8.0 does work. I've reinstalled all my extensions but this one. This one is the only one that is giving me grief about reinstalling.

I looked at the installer log, and supported version is 'Version : [14.0,18.0)', so it should be allowed... I'm inclined to believe it's still MS's fault that I can't install your extension, but seeing as all other extensions I use were able to be installed, I thought I'd at least post the log here, and see if there is anything I can do to fix this? I'm going to and jump a minor version or two forward and see if that helps, but it's such a tedious process to do that with VS.

Log file:
[dd_VSIXInstaller_20210721112728_24dc.log](https://github.com/gitextensions/gitextensions.vsix/files/6857020/dd_VSIXInstaller_20210721112728_24dc.log)

**Comment by BOBJohnson** [comment]
Welp, I've upgraded to 16.9.0 and still can't install it. I've also gone to 16.10.0 and can't install it. Is there some kind of VSIX cache that I can clear?

**Comment by BOBJohnson** [comment]
Ugh. I just upgraded to 16.10.4 and somehow even though I have repeatedly re-installed the latest version of VS 2019 (and I did a Windows refresh Monday) it now loads the projects perfectly.

But now I can't install your extension at all. I must be missing a VS component maybe? Any idea what component I might be missing?

**Comment by BOBJohnson** [comment]
GAH!

I just noticed this in the log:

7/21/2021 12:05:37 PM - vsixinstaller.exe version:
7/21/2021 12:05:37 PM - **17.0.2145-preview2**

I installed VS 2022 yesterday in an effort to see if I could get back to working that way. So this is VS's fault. Now I have to figure out how to rollback the vsixinstaller.exe even though I've already uninstalled VS 2022.

**Comment by BOBJohnson** [comment]
Just in-case someone else happens upon the same bizarre circumstances where you have a Preview version of VS that is preventing you from installing legitimate extensions here was my solution:

- Navigate to: C:\Program Files (x86)\Microsoft Visual Studio\
- Remove or Rename the 'Installer' folder. Apparently there is no Uninstall for the Installer, and this is all you need to do to remove it.
- Launch a installer exe from your downloads folder that isn't for the Preview installer. This will reinstall the older installer.
- Launch the VSIX.

**Comment by RussKie** [comment]
VS 2022 uses a completely different extensibility model, so VSIXs built for previous versions won't work in it.

**Comment by BOBJohnson** [comment]
> 
> 
> VS 2022 uses a completely different extensibility model, so VSIXs built for previous versions won't work in it.

My issue was that I had uninstalled VS 2022, and had installed VS 2019, but the VS 2022 Installer got left behind, and even though I was trying to load an extension in VS 2019, it was using the VS 2022 VSIX installer instead of the VS 2019 VSIX installer.

I don't know if this was your issue, or the VS 2022 VSIX installer exe's issue, but your extension was the only extension that wouldn't install to VS 2019 with the VS 2022 VSIX Installer exe. I installed several other VS 2019 extensions and they all installed correctly.
